Main.py

```diff
@@ -6,130 +6,186 @@
 from PyQt5.QtGui import *
 from PyQt5.QtWidgets import *
 from PySide2 import QtCore
-import glob
 import subprocess
 import requests
 import os
 import sys
+import shutil
+import pathlib
 
 # set path
 def resource_path(relative_path):
-    base_path = os.path.abspath("./resources")
-    return os.path.join(base_path, relative_path)
+    return os.path.join(cwd, "resources", relative_path)
+
+
+cwd = os.path.dirname(os.path.abspath(__file__))
+if shutil.which("altserver"):
+    AltServer = shutil.which("altserver")
+    using_resource_altserver = False
+else:
+    AltServer = resource_path("AltServer")
+    using_resource_altserver = True
 
-cwd = os.getcwd()
-AltServer = resource_path("AltServer")
 AutoStart = resource_path("AutoStart.sh")
-Exec = cwd+"/altserver"
+Exec = cwd + "/altserver"
 UserName = os.getlogin()
 
+HOME = os.path.expanduser("~")
+TMP_FILE = os.path.join(HOME, ".tmp/log.txt")
+if not os.path.exists(TMP_FILE):
+    os.makedirs(os.path.dirname(TMP_FILE), exist_ok=True)
+    pathlib.Path(TMP_FILE).touch()
+
+
 def internet_stat():
     timeout = 5
     try:
-        requests.get("https://github.com", timeout=timeout)
+        requests.get("https://example.com", timeout=timeout)
```

Comment:
I prefer to using github.com , if github website down,them we can't use update service

Reply:
In the latest commit, I've switched back to github.com, I was thinking spamming github probably wasn't the best idea